api/filter_outputs.go

```diff
@@ -1,14 +1,63 @@
 package api
 
 import (
+	"fmt"
 	"net/http"
+	"strings"
 
 	"github.com/ONSdigital/dp-cantabular-filter-flex-api/model"
 	"github.com/ONSdigital/log.go/v2/log"
 	"github.com/go-chi/chi/v5"
 	"github.com/pkg/errors"
 )
 
+func (api *API) getFilterOutput(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	fID := chi.URLParam(r, "filter-output-id")
+
+	var filterOutput *model.FilterOutput
+
+	filterOutput, err := api.store.GetFilterOutput(ctx, fID)
+	if err != nil {
+
+		status := http.StatusNotFound
+		message := "filter output not found"
+
```

Comment:
Ok this stuff is a little unnecessary, if you look at some of the existing mongo functions you'll see we add tags like `notFound` etc to the errors there, then here we use the statusCode(err) function to determine which statusCode to return. The default will be 500. So here you can just do:
```
if err != nil{
  		api.respond.Error(
			ctx,
			w,
			statusCode(err),
			Error{
				err: errors.Wrap(err, "failed to get filter output),
				logData: log.Data{
					"id": fID,
				},
			},
		)
}
```

Reply:
Also in general it's considered bad practice to inspect the error strings to determine what to do with errors. It's better to use interfaces to determine behaviour (as we have set up, you'll see more further down)

Reply:
@ChrysmOre  understood and noted. The recent commit checks with `statusCode(err)` as you ask. However, the error codes are now 500 for both of the failed cases that I have tested, which I assume is okay. I ammended the tests to reflect this. 

Also I believe the use of `statusCode(err)` isn't consistent in some of the methods, where instead `http.Status<codetype>` is used instead. This probably can be cleaned up in the future.

Reply:
@ChrysmOre this has been resolved. `statusCode(err)` is now being used, with 404s and 500s returned.